libcron/include/libcron/Cron.h

```diff
@@ -139,6 +140,19 @@ namespace libcron
             bool first_tick = true;
             std::chrono::system_clock::time_point last_tick{};
     };
+    
+    template<typename ClockType, typename LockType>
+    bool Cron<ClockType, LockType>::was_executed_on_time(const std::string& name)
+    {
+        tasks.lock_queue();
+        std::vector<Task>& tvec = tasks.get_tasks();
+        auto it = std::find_if(tvec.begin(), tvec.end(), [&name](const Task& T){return T.get_name() == name; });
+        tasks.release_queue();
+        if (it != tvec.end())
+            return it->executed_on_time();
+        else
+            return false;
```

Comment:
This isn't thread-safe. There's no guarantee that `tvec` isn't modified once you have released the queue so the iterators you got from it may be invalidated.

If possible, please don't do multiple returns. `return it != tvec.end() && it->executed_on_time();` (though as per the above, the it may not be valid here). Also, always use {} for branches. :)

Reply:
Jup, isn't thread safe. I added a test to ensure functionality for Mt-environements as well.
And of course, a good idea from your side. I will adapt the code. 

PS: I try to stick to your coding guidelines, thanks in advance for your tips! Making my style better as well.

Reply:
I've not set up CI for this project yet as there have been no need to coordinate contributors yet, but feel free to have a look at [this guideline](https://github.com/PerMalmberg/Smooth/blob/master/CONTRIBUTING.md#formatting--style) and the uncrustify config I use my other project. (I'm sure this one is not 100% compliant yet)